### PR TITLE
Update CPU affinity parameter to be able to set multiple CPUs

### PR DESCRIPTION
**Pull request by saikishor**
This PR changes the strategy on handling CPU affinity. The changes here allow to use multiple CPUs instead of a single one as in previous implementation

Similar approach as https://github.com/ros-controls/ros2_control/pull/1898

### CODE REVIEW

**Comment by saikishor** [comment]
@firesurfer Can you take a look at the changes?

File: controller_manager/src/ros2_control_node.cpp [hunk 1]
```diff
@@ -67,10 +67,22 @@ int main(int argc, char ** argv)
...
+  rclcpp::Parameter cpu_affinity_param;
+  if (cm->get_parameter("cpu_affinity", cpu_affinity_param))
   {
-    const auto affinity_result = realtime_tools::set_current_thread_affinity(cpu_affinity);
+    std::vector<int> cpus = {};
+    if (cpu_affinity_param.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER)
+    {
+      cpus = {cpu_affinity_param.as_int()};
+    }
+    else if (cpu_affinity_param.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER_ARRAY)
```
Comment (firesurfer) [inline controller_manager/src/ros2_control_node.cpp:78]: You might want to update the documentation: https://github.com/ros-controls/ros2_control/blob/master/controller_manager/doc/userdoc.rst

**Review by firesurfer — APPROVED** [review]
Apart from the missing documentation update this looks fine!

File: controller_manager/src/ros2_control_node.cpp (hunk 1 above)
Comment (saikishor) [reply controller_manager/src/ros2_control_node.cpp:78]: Thank you for reminding!

**Review by firesurfer — APPROVED** [review]

**Review by christophfroehlich — COMMENTED** [review]
Should we update that in the release [notes](https://github.com/ros-controls/ros2_control/blob/master/doc/release_notes.rst#controller_manager)?

**Comment by saikishor** [comment]
> Should we update that in the release [notes](https://github.com/ros-controls/ros2_control/blob/master/doc/release_notes.rst#controller_manager)?

Yes! You are right. I forgot that part 😅😅

**Review by christophfroehlich — APPROVED** [review]
Thanks!

**Review by bmagyar — APPROVED** [review]

**Comment by christophfroehlich** [comment]
@mergifyio backport humble

**Pull request merged**